### PR TITLE
ceph.spec: fix mgr diskprediction_local dependency

### PR DESCRIPTION
**Pull request by ideepika**
fixes: https://tracker.ceph.com/issues/44948
Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: ceph.spec.in [hunk 1]
```diff
@@ -563,9 +563,11 @@ Group:          System/Filesystems
 Requires:       ceph-mgr = %{_epoch_prefix}%{version}-%{release}
 Requires:       python%{python3_pkgversion}-numpy
 Requires:       python3-scipy
+Requires:       python%{python3_pkgversion}-scikit-learn
 %if 0%{?rhel} == 7
 Requires:       numpy
 Requires:       scipy
+Requires:       scikit-learn
```
Comment (tchaikov) [inline ceph.spec.in:570]: i don't think el7 or EPEL7 has this package.

Comment (ideepika) [reply ceph.spec.in:570]: yes, thanks @tchaikov I will add it at container side for rhel7, doesn't look like we can do much here.

Comment (jdurgin) [reply ceph.spec.in:570]: now that we're not building master on el7, does this matter?

Comment (tchaikov) [reply ceph.spec.in:570]: probably we can go further by removing the el7 stuff in `ceph.spec.in`.

Comment (ideepika) [reply ceph.spec.in:570]: I was going to ask the same question, do we require it for previous ceph versions, or should we clean up el7 stuff?

Comment (tchaikov) [reply ceph.spec.in:570]: i'd suggest just drop the change for el7. and do the cleanup in a separate PR.

Comment (ideepika) [reply ceph.spec.in:570]: ack. Thanks!

File: ceph.spec.in [hunk 2]
```diff
@@ -563,6 +563,7 @@ Group:          System/Filesystems
 Requires:       ceph-mgr = %{_epoch_prefix}%{version}-%{release}
 Requires:       python%{python3_pkgversion}-numpy
 Requires:       python3-scipy
+Requires:       python%{python3_pkgversion}-scikit-learn
```
Comment (tchaikov) [inline ceph.spec.in:566]: one last nit, could you move this line before `python3-scipy`, so they are ordered alphabetically.

Comment (ideepika) [reply ceph.spec.in:565]: ah missed it, updated!

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2020-06-25_08:00:18-rados-wip-kefu-testing-2020-06-25-1439-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/42390/console

**Pull request merged**